src/sisl/physics/electron.py

```diff
@@ -1571,9 +1579,41 @@ def norm2(self, sum=True):
         # Retrieve the overlap matrix (FULL S is required for NC)
         S = self.Sk()
 
-        if sum:
+        # ensure backwards compatibility by converting bool
+        if projection is True:
+            projection = "sum"
+        elif projection is False:
+            projection = "orbital"
+
+        if projection == "sum":
             return self.inner(matrix=S)
-        return conj(self.state) * S.dot(self.state.T).T
+
+        elif projection == "orbital":
+            return conj(self.state) * S.dot(self.state.T).T
+
+        elif projection == "atom":
+            na = self.parent.na
+            no = self.parent.no
+
+            # build sparse matrix M to map from orbital-to-atom-resolved quantity
+            idx = self.parent.a2o(range(na))
```

Comment:
I think it is faster to do:
```python
col_idx = np.arange(len(self.parent))
row_idx = self.parent.o2a(col_idx)
```
isn't this the same?

Reply:
Yes, much simpler :)